From deadffda27b9b9eae0ffa819a633e4cf81ee22f4 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Fri, 2 Jun 2017 14:01:17 -0400
Subject: [PATCH 7/7] v4l2videoenc: Make sure min_buffers is valid

When upstream does no use the v4l2videoenc pool, we need to activate
that internal pool. Though, we relied the driver to provide a minimum
required buffer, which Qualcomm Venus driver don't currently provide.

https://bugzilla.gnome.org/show_bug.cgi?id=783361
(cherry picked from commit b9532fc6fb94293505cf14441f14009cf9250975)
---
 sys/v4l2/gstv4l2videoenc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index f66744be5..8c432105a 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -512,9 +512,10 @@ gst_v4l2_video_enc_handle_frame (GstVideoEncoder * encoder,
     /* Ensure input internal pool is active */
     if (!gst_buffer_pool_is_active (pool)) {
       GstStructure *config = gst_buffer_pool_get_config (pool);
+      guint min = MAX (self->v4l2output->min_buffers, GST_V4L2_MIN_BUFFERS);
+
       gst_buffer_pool_config_set_params (config, self->input_state->caps,
-          self->v4l2output->info.size, self->v4l2output->min_buffers,
-          self->v4l2output->min_buffers);
+          self->v4l2output->info.size, min, min);
 
       /* There is no reason to refuse this config */
       if (!gst_buffer_pool_set_config (pool, config))
-- 
2.13.3

